### ISSUE TITLE
`create` option not honored

### ISSUE THREAD

**Issue by geoffjukes** [althonos/fs.sshfs#10]
Hi,

If I pass the `create` option to an SFTP endpoint, it fails with `fs.errors.ResourceNotFound`. The host exists, and I can `makedirs` after connecting to the root.

For example:

```
import fs
sftp = fs.open_fs('sftp://example.com/path/to/save', create=True)
```
raises `fs.errors.ResourceNotFound`
```
import fs
root = fs.open_fs('sftp://example.com')
sftp = root.makedirs('/path/to/save')
```
Works just fine.

Thoughts?

Geoff

**Comment by althonos** [comment]
Hi @geoffjukes,
I indeed did not implement the case of the `create` argument (and I also didn't implement it in `fs.smbfs`, I'll fix it as well).

In 48664f2251e1d5c80dc4cf9bb36014a3cde3880d I made the following fixes:
* if `create` is set to `True`, `ssh_fs.makedirs(dir_path, recreate=True)` will be called before `opendirs`
* I wrapped the call to `opendir` and `makedirs` within a `try/except` block so that any exception will be catched and then reraised as a `CreateFailed` exception.

**Comment by althonos** [comment]
Fixed in `v0.7.2`. Thanks for the report !
